packages/component/src/hooks/internal/useFocusWithin.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import type { RefObject } from 'react';
+
+export default function useFocusWithin(targetRef: RefObject<HTMLElement>): readonly [boolean] {
+  const [focusWithin, setFocusWithin] = useState(false);
```

Comment:
It's not perfect for this scenario, but you should consider adding a `useBoolean` hook to Web Chat. I find it super convenient in PVA.

Reply:
Will file a feature request and we change many of them at once. And I will tag you as a reviewer. 🤣